tensorboard/plugins/pr_curve/pr_curves_plugin.py

```diff
@@ -0,0 +1,198 @@
+
+
+class PrCurvesPlugin(base_plugin.TBPlugin):
+  """A plugin that serves PR curves for individual classes."""
+
+  plugin_name = metadata.PLUGIN_NAME
+
+  def __init__(self, context):
+    """Instantiates a PrCurvesPlugin.
+    Args:
+      context: A base_plugin.TBContext instance. A magic container that
+        TensorBoard uses to make objects available to the plugin.
+    """
+    self._multiplexer = context.multiplexer
+
+  @wrappers.Request.application
+  def pr_curves_route(self, request):
+    """A route that returns a JSON mapping between runs and PR curve data.
+
+    Returns:
+      Given a tag and a comma-separated list of runs (both stored within GET
+      parameters), fetches a JSON object that maps between run name and objects
+      containing data required for PR curves for that run. Runs that either
+      cannot be found or that lack tags will be excluded from the response.
+    """
+    runs = request.args.getlist('run')
+    if not runs:
+      return http_util.Respond(
+          request, 'No runs provided when fetching PR curve data', 400)
+
+    tag = request.args.get('tag')
+    if not tag:
+      return http_util.Respond(
+          request, 'No tag provided when fetching PR curve data', 400)
+
+    try:
+      response = http_util.Respond(
+          request, self.pr_curves_impl(runs, tag), 'application/json')
+    except ValueError as e:
+      return http_util.Respond(request, '%s' % e, 'text/plain', 400)
+
+    return response
+
+  def pr_curves_impl(self, runs, tag):
+    """Creates the JSON object for the PR curves response for a run-tag combo.
+
+    Arguments:
+      runs: A list of runs to fetch the curves for.
+      tag: The tag to fetch the curves for.
+
+    Raises:
+      ValueError: If no PR curves could be fetched for a run and tag.
+
+    Returns:
+      The JSON object for the PR curves route response.
+    """
+    response_mapping = {}
+    for run in runs:
+      try:
+        tensor_events = self._multiplexer.Tensors(run, tag)
+      except KeyError:
+        raise ValueError(
+            'No PR curves could be fetched for run %r and tag %r' % (run, tag))
+
+      response_mapping[run] = [
+          self._process_tensor_event(e) for e in tensor_events]
+    return response_mapping
+
+  @wrappers.Request.application
+  def tags_route(self, request):
+    """A route (HTTP handler) that returns a response with tags.
+    Returns:
+      A response that contains a JSON object. The keys of the object
+      are all the runs. Each run is mapped to a (potentially empty)
+      list of all tags that are relevant to this plugin.
+    """
+    return http_util.Respond(
+        request, self.tags_impl(), 'application/json')
+
+  def tags_impl(self):
+    """Creates the JSON object for the tags route response.
+
+    Returns:
+      The JSON object for the tags route response.
+    """
+    all_runs = self._multiplexer.PluginRunToTagToContent(
+        PrCurvesPlugin.plugin_name)
```

Comment:
This should be `metadata.PLUGIN_NAME`. I know that they have the same value, but the argument to `PluginRunToTagToContent` corresponds to the string stored in the `plugin_data` of the summaries, not the route used to fetch data. If you changed `PrCurvesPlugin.plugin_name` to be something else, the value of this argument should still be the original `metadata.PLUGIN_NAME`.

Likewise elsewhere in this file.